### PR TITLE
caddy: use darccio/mergo, fixing github stealth update

### PR DESCRIPTION
**Pull request by mrdomino**
#### Description

Closes: https://trac.macports.org/ticket/69132

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 14.2.1 23C71 arm64
Xcode 15.2 15C500b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by mrdomino** [comment]
Does not build:

```
:info:build Command failed:  cd "/opt/local/var/macports/build/_Users_joshin_macports-ports_www_caddy/caddy/work/gopath/src/github.com/caddyserver/caddy/cmd/caddy" && /opt/local/bin/go build 
:info:build Exit code: 1
:error:build Failed to build caddy: command execution failed
:debug:build Error code: CHILDSTATUS 4439 1
:debug:build Backtrace: command execution failed
:debug:build     while executing
:debug:build "system {*}$notty {*}$callback {*}$nice $fullcmdstring"
:debug:build     invoked from within
:debug:build "command_exec -callback portprogress::target_progress_callback build"
:debug:build     (procedure "portbuild::build_main" line 8)
:debug:build     invoked from within
:debug:build "$procedure $targetname"
:error:build See /opt/local/var/macports/logs/_Users_joshin_macports-ports_www_caddy/caddy/main.log for details.
```

File: www/caddy/Portfile [hunk 1]
```diff
@@ -550,11 +550,11 @@ go.vendors          howett.net/plist \
                         rmd160  88f9577df93ac0f8801d8960adc7f28e38867f3e \
                         sha256  f69af10ff08c0e87f92dac3ee5172d8ed02463725b74edfc8943ef018a1a632d \
                         size    5343 \
-                    github.com/imdario/mergo \
```
Comment (mohd-akram) [inline www/caddy/Portfile:553]: Keep the same name and add `repo github.com/darccio/mergo` after this.

Comment (mrdomino) [reply www/caddy/Portfile:553]: Looks like that works!

**Comment by mohd-akram** [comment]
If this works, then could you drop the version back to `v0.3.12` (that's what it was upstream, I only bumped it to possibly fix the issue) and bump the port revision as well.

**Comment by mrdomino** [comment]
Done.

**Comment by mrdomino** [comment]
Arguably a revision bump is unnecessary here, since the code is identical, just the base directory is different. But I don't care either way.

**Comment by mohd-akram** [comment]
That would be the case if I hadn't changed the version to v0.3.13 :D

**Review by mohd-akram — APPROVED** [review]

**Comment by mrdomino** [comment]
ohhhhh right

**Comment by mrdomino** [comment]
Most recent push just changes commit message.

**Comment by mrdomino** [comment]
Anything else required to merge this?

**Comment by mohd-akram** [comment]
Nope, thank you for the PR! I can't merge it myself as I don't have the rights.

**Comment by mrdomino** [comment]
Picking a name at random based on recent helpfulness: @pmetzger?

**Pull request merged**